tests/test_inner.py

```diff
@@ -723,7 +769,6 @@ def test_workflow_plugin_error(fail_at):
     # an exit plugin that's explicitly allowed to fail.
     if fail_at == 'exit_raises_allowed':
         build_result = workflow.build_docker_image()
-        assert build_result and not build_result.is_failed()
```

Comment:
What does workflow.build_docker_image() return now, nothing? It just raises an exception on error? If that's the case we shouldn't assign to build_result.